Parcial 1 Minería de datos

> Daniela Alexandra Herrera Fandiño

In [1]:
import pandas as pd
import numpy as np
import csv

In [2]:
pd.set_option('display.max_columns', None) #Mostrar todas las columnas


In [3]:
df = pd.read_csv(
    'https://raw.githubusercontent.com/prof-tatiana-gutierrez/DatosMD/master/student-mat.csv',
    sep = ";")
df.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher   
1     GP   F   17       U     GT3       T     1     1  at_home     other   
2     GP   F   15       U     LE3       T     1     1  at_home     other   
3     GP   F   15       U     GT3       T     4     2   health  services   
4     GP   F   16       U     GT3       T     3     3    other     other   

   reason guardian  traveltime  studytime  failures schoolsup famsup paid  \
0  course   mother           2          2         0       yes     no   no   
1  course   father           1          2         0        no    yes   no   
2   other   mother           1          2         3       yes     no  yes   
3    home   mother           1          3         0        no    yes  yes   
4    home   father           1          2         0        no    yes  yes   

  activities nursery higher internet romantic  famrel  freetime  goout  Dalc  \
0         no     yes    yes       no       no       4         3      4     1   
1         no      no    yes      yes       no       5         3      3     1   
2         no     yes    yes      yes       no       4         3      2     2   
3        yes     yes    yes      yes      yes       3         2      2     1   
4         no     yes    yes       no       no       4         3      2     1   

   Walc  health  absences  G1  G2  G3  
0     1       3         6   5   6   6  
1     1       3         4   5   5   6  
2     3       3        10   7   8  10  
3     1       5         2  15  14  15  
4     2       5         4   6  10  10

In [4]:
df.isna().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

In [5]:
df.isnull().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

In [6]:
pd.unique(df['address'])

array(['U', 'R'], dtype=object)

## Analizando faltantes

In [9]:
#pd.isnull(df)
for col in df.columns:
    faltantes = df[col].isnull().sum()
    pct_faltantes = np.mean(df[col].isnull())
    print('{} - {}% - {}'.format(col,round(pct_faltantes*100, 2),faltantes))

school - 0.0% - 0
sex - 0.0% - 0
age - 0.0% - 0
address - 0.0% - 0
famsize - 0.0% - 0
Pstatus - 0.0% - 0
Medu - 0.0% - 0
Fedu - 0.0% - 0
Mjob - 0.0% - 0
Fjob - 0.0% - 0
reason - 0.0% - 0
guardian - 0.0% - 0
traveltime - 0.0% - 0
studytime - 0.0% - 0
failures - 0.0% - 0
schoolsup - 0.0% - 0
famsup - 0.0% - 0
paid - 0.0% - 0
activities - 0.0% - 0
nursery - 0.0% - 0
higher - 0.0% - 0
internet - 0.0% - 0
romantic - 0.0% - 0
famrel - 0.0% - 0
freetime - 0.0% - 0
goout - 0.0% - 0
Dalc - 0.0% - 0
Walc - 0.0% - 0
health - 0.0% - 0
absences - 0.0% - 0
G1 - 0.0% - 0
G2 - 0.0% - 0
G3 - 0.0% - 0


No existen valores nulos o faltantes en ninguna de las variables que contiene el datset

## Tipo de domicilio

In [17]:
df_add = df['address']
df_add

0      U
1      U
2      U
3      U
4      U
      ..
390    U
391    U
392    R
393    R
394    U
Name: address, Length: 395, dtype: object

In [63]:
(df.groupby("address")
  .agg(frequency=("address", "count")))

frequency
address           
R               88
U              307

Se encuentran 88 estudiantes los cuales viven en zona rural y 307 estudiantes habitan en zona urbana

In [74]:
df_add2 = df[(df['address'] == 'U')]
df_add2[df_add2(df['sex'] == 'M')]


df_mhd2 = df_mhd[(df_mhd['Year'] == 2000)]
df_mhd2[df_mhd2['Code'].isnull()]

TypeError: ignored

In [54]:
# numero de fallas 
#df_abs = df['absences']
df_abs[df_abs > 5]
df_abs

0       6
1       4
2      10
3       2
4       4
       ..
390    11
391     3
392     3
393     0
394     5
Name: absences, Length: 395, dtype: int64

In [44]:
#internet
df_int = df['internet']
df_int

0       no
1      yes
2      yes
3      yes
4       no
      ... 
390     no
391    yes
392     no
393    yes
394    yes
Name: internet, Length: 395, dtype: object